python/tskit/text_formats.py

```diff
@@ -0,0 +1,85 @@
+# MIT License
+#
+# Copyright (c) 2021 Tskit Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+Module responsible for working with text format data.
+"""
+import numpy as np
+
+import tskit
+
+
+def parse_fam(fam_path: str, tc: tskit.TableCollection):
+    """
+    Parse PLINK .fam file and convert to tskit IndividualTable.
+
+    Assumes fam file contains five columns: FID, IID, PAT, MAT, SEX
+
+    :param str fam_path: Path to PLINK .fam file
+    :param tskit.TableCollection tc: TableCollection with IndividualTable to
+        which the individuals will be added
+    """
+    individuals = np.loadtxt(
+        fam_path,
+        dtype=str,
+        ndmin=2,  # read file as 2-D table
+        usecols=(0, 1, 2, 3, 4),  # only keep FID, IID, PAT, MAT, SEX columns
+    )  # requires same number of columns in each row, i.e. not ragged
+
+    if len(individuals) == 0:
+        return tc
+
+    id_map = {}  # dict for translating PLINK ID to tskit IndividualTable ID
+    for tskit_id, (plink_fid, plink_iid, _pat, _mat, _sex) in enumerate(individuals):
+        # need this to guarantee uniqueness of individual ID
+        plink_id = (
+            str(plink_fid) + " " + str(plink_iid)
+        )  # include space between strings to ensure uniqueness
+        if plink_id in id_map:
+            raise ValueError("Duplicate PLINK ID")
+        id_map[plink_id] = tskit_id
+    id_map["0"] = "-1"
+
+    tb = tc.individuals
+    tb.metadata_schema = tskit.MetadataSchema(
+        {
+            "codec": "json",
+            "type": "object",
+            "properties": {
```

Comment:
I'd be tempted to put properties `fid`,`iid` and `sex` under a `plink` object.

Reply:
how would I do this?

Reply:
Let's keep it flat for now - I think we'll need to think hard about the metadata we want to support for individuals, and what we have now almost certainly isn't it. This will get us going, though.